pkg/controller/volume/attachdetach/reconciler/reconciler_test.go

```diff
@@ -1594,15 +1593,23 @@ func verifyVolumeAttachedToNode(
 	expectedAttachState cache.AttachState,
 	asw cache.ActualStateOfWorld,
 ) {
-	attachState := asw.GetAttachState(volumeName, nodeName)
-	if attachState != expectedAttachState {
-		t.Fatalf("Check volume <%v> is attached to node <%v>, got %v, expected %v",
-			volumeName,
-			nodeName,
-			attachState,
-			expectedAttachState)
+	var lastErr error
+	err := wait.PollUntilContextTimeout(context.TODO(), 50*time.Millisecond, 2*time.Second, true, func(context.Context) (done bool, err error) {
```

Comment:
if @msau42 confirms polling is correct, a 2 second timeout is too short / aggressive for a unit test ... anything that could take O(seconds) could take 10+ seconds in resource-constrained CI environments

Reply:
also, use one of the PollImmediateUntil__ variants so we check immediately rather than waiting the interval the first time?

Reply:
Polling of `attachState` should not be necessary though. The volume isn't removed from attachState and hence that state should never change. If anything - if it does changes we probably have a bug..

Reply:
Then I would like to remove this polling here.